step-automation-packages/step-automation-packages-controller/src/main/java/step/automation/packages/AutomationPackageServices.java

```diff
@@ -136,22 +135,23 @@ true, false, new ObjectId(id),
 
     @PUT
     @Consumes(MediaType.MULTIPART_FORM_DATA)
-    @Produces(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_JSON)
     @Secured(right = "automation-package-write")
-    public Response createOrUpdateAutomationPackage(@FormDataParam("file") InputStream uploadedInputStream,
+    public Response createOrUpdateAutomationPackage(@QueryParam("async") Boolean async,
+                                                    @FormDataParam("file") InputStream uploadedInputStream,
                                                     @FormDataParam("file") FormDataContentDisposition fileDetail) {
         try {
-            AutomationPackageManager.PackageUpdateResult result = automationPackageManager.createOrUpdateAutomationPackage(
+            AutomationPackageUpdateResult result = automationPackageManager.createOrUpdateAutomationPackage(
                     true, true, null, uploadedInputStream, fileDetail.getFileName(),
-                    getObjectEnricher(), getObjectPredicate()
+                    getObjectEnricher(), getObjectPredicate(), async != null && async
             );
             Response.ResponseBuilder responseBuilder;
-            if (result.getStatus() == AutomationPackageManager.PackageUpdateStatus.CREATED) {
+            if (result.getStatus() == AutomationPackageUpdateStatus.CREATED) {
                 responseBuilder = Response.status(201);
```

Comment:
Consider using Response.Status.OK and Response.Status.CREATED instead of raw magic numbers :)